### PR TITLE
Create standalone go module for addon config apis

### LINKED ISSUE #3225: Refactor cpi,cni,csi apis to its own go module
**Describe the feature request**
As part of tanzu-framework refactoring we are breaking up repo into multiple go modules. Move cpi, cni and csi addons config apis to a standalone go module.

**Describe alternative(s) you've considered**
N/A
**Affected product area (please put an X in all that apply)**

- (x) APIs
- (x) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

### PR DESCRIPTION
**Pull request by vijaykatam**
Signed-off-by: Vijay Katam <vkatam@vmware.com>

### What this PR does / why we need it

As part of tanzu-framework refactoring we are breaking up repo into multiple go modules. This change creates standalone go module for addon config apis

### Which issue(s) this PR fixes

Fixes #3225

### Describe testing done for PR
Unit tests and env tests passing

### Release note

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -150,8 +150,6 @@ deploy: manifests tools ## Deploy controller in the configured Kubernetes cluste
 
 manifests: ## Generate manifests e.g. CRD, RBAC etc.
 	$(MAKE) generate-manifests CONTROLLER_GEN_SRC=./apis/...
-	$(MAKE) -C apis/cpi generate-manifests CONTROLLER_GEN_SRC=./...
-	$(MAKE) -C apis/cni generate-manifests CONTROLLER_GEN_SRC=./...
```
Comment (danniel1205) [inline Makefile:154]: question: wondering if still need those code-gen manifests-gen stuff for cpi and cni ? Like `-C apis/addonconfigs/cni` ?

File: apis/addonconfigs/hack/boilerplate.go.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Copyright YEAR VMware, Inc. All Rights Reserved.
```
Comment (danniel1205) [inline apis/addonconfigs/hack/boilerplate.go.txt:1]: question: do we need this file ? Or maybe `YEAR` --> `2022` ?

File: Makefile (hunk 1 above)
Comment (danniel1205) [reply Makefile:154]: nvm, I see a new makefile gets created for cni and cpi under `apis/addonconfigs`

File: apis/addonconfigs/hack/boilerplate.go.txt (hunk 2 above)
Comment (vijaykatam) [reply apis/addonconfigs/hack/boilerplate.go.txt:1]: This is the boilerplate template copied from root needed to keep the module standalone. `YEAR` get substituted by Makefile target.

**Review by danniel1205 — APPROVED** [review]

**Pull request merged**